### PR TITLE
feat(Prompter): Support for multiple Joycon controller for prompter

### PR DESCRIPTION
**Pull request by jesperstarkar**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
New feature/enhanced capatibility

* **What is the current behavior?** (You can also link to an open issue here)
Only a singe L or R or LR(pair) joycon(s) can be connected.
Known problems when going from LR mode to either L or R, where input might get stuck until windows restart.

* **What is the new behavior (if this is a feature change)?**
Multiple LL or RR controllers supported. Less/no problems when going LR to L or R. 

* **Other information**:

**Status**

- [ ] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [ ] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

**Review by jstarpl — CHANGES_REQUESTED** [review]
Overall, seems good, one line seems like it can be removed, one question about typings.

File: meteor/client/ui/Prompter/controller/joycon-device.ts [hunk 1]
```diff
@@ -208,75 +201,94 @@ export class JoyConController extends ControllerAbstract {
...
-		if (this.lastUsedJoyconMode === 'L' || this.lastUsedJoyconMode === 'R') {
-			// L or R mode
-			if (Math.abs(input.axes[0]) > this.deadBand) {
-				if (this.lastUsedJoyconMode === 'L') {
-					return input.axes[0] * -1 // in this mode, L is "negative"
-				} else if (this.lastUsedJoyconMode === 'R') {
-					return input.axes[0] * 1.4 // in this mode, R is "positive"
+			} else if (joycon.mode === 'LR') {
+				// L + R mode
+				// get the first one that is moving outside of the deadband, priorotizing the L controller
```
Comment (baltedewit) [inline meteor/client/ui/Prompter/controller/joycon-device.ts:240]: Behaviour appears to be incorrect compared to this comment. Currently the right one will overwrite left if it moves out of the deadband.

**Review by baltedewit — CHANGES_REQUESTED** [review]

File: meteor/client/ui/Prompter/controller/joycon-device.ts [hunk 2]
```diff
@@ -2,7 +2,8 @@ import { ControllerAbstract } from './lib'
 import { PrompterConfigMode, PrompterViewInner } from '../PrompterView'
 import Spline from 'cubic-spline'
 
-type JoyconMode = 'L' | 'R' | 'LR'
+type JoyconWithData = { index: number; mode: JoyconMode; axes: readonly number[]; buttons: number[] }
```
[the remark this answers is not in the document]
Comment (jesperstarkar) [reply meteor/client/ui/Prompter/controller/joycon-device.ts:5]: @jstarpl it comes from the standard typings for gamepads - I have not invented this.

TypeScript says that the number we are reading from is readonly, and requires that when we assign it we keep that type. I just did so, and didn't think more about it. I don't want to add stricter types than needed, neither do I want to cast or work around this warning.

Agree?

image

File: meteor/client/ui/Prompter/controller/joycon-device.ts (hunk 1 above)
Comment (jesperstarkar) [reply meteor/client/ui/Prompter/controller/joycon-device.ts:240]: Fixed by adding the second clause in elseif instead of another if.

**Pull request merged**